substrate/client/service/src/client/client.rs

```diff
@@ -326,19 +325,35 @@ where
 			// dropped, the block will be unpinned automatically.
 			if let Some(ref notification) = finality_notification {
 				if let Err(err) = self.backend.pin_block(notification.hash) {
-					error!(
+					debug!(
 						"Unable to pin block for finality notification. hash: {}, Error: {}",
 						notification.hash, err
 					);
-				};
+				} else {
+					let _ = self
+						.unpin_worker_sender
+						.unbounded_send(UnpinWorkerMessage::AnnouncePin(notification.hash))
+						.map_err(|e| {
+							log::debug!(
+								"Unable to send AnnouncePin worker message for finality: {e}"
+							)
+						});
```

Comment:
Why if we fail to `pin_block` we don't consider this an `error` anymore?
Also what if the `upbounded_send` fails? Shouldn't this be logged as an error? Is ok to ignore the error here?

Reply:
When doing fast-sync it happens that we receive finality notifications for blocks that don't have state. We had an issue abouit these error messages causing confusion, so its debug now.

Made the unbounded send an error  however, thats a good point :+1: